src-java/kilda-model/src/main/java/org/openkilda/model/HaFlow.java

```diff
@@ -373,6 +374,44 @@ public HaSubFlow getHaSubFlowOrThrowException(String haSubFlowId) {
                 format("HA-flow %s has no HA-sub flow %s", getHaFlowId(), haSubFlowId)));
     }
 
+    /**
+     * Swap primary and protected path IDs.
+     */
+    public void swapPathIds() {
+        final PathId primaryForward = getForwardPathId();
+        final PathId primaryReverse = getReversePathId();
+        final PathId protectedForward = getProtectedForwardPathId();
+        final PathId protectedReverse = getProtectedReversePathId();
+
+        setForwardPathId(protectedForward);
+        setReversePathId(protectedReverse);
+        setProtectedForwardPathId(primaryForward);
+        setProtectedReversePathId(primaryReverse);
+    }
+
+    public Collection<HaFlowPath> getPrimaryPaths() {
+        return getHaFlowPaths(getForwardPath(), getReversePath());
+    }
+
+    public Collection<HaFlowPath> getProtectedPaths() {
+        return getHaFlowPaths(getProtectedForwardPath(), getProtectedReversePath());
+    }
+
+    /**
+     * Returns HA-flow paths which are currently in use.
+     * This method doesn't return unused paths, like getPath() method does.
+     * Paths can be unused in the middle of update/reroute operations.
```

Comment:
Feels like something is missing here. Maybe "paths can be **set** unused ...". Why it is important? How update/reroute is connected to this method?

Reply:
I left this comment just to indicate difference between getPaths() and this method. 

Comment was fixed